tdm_loader/tdm_loader.py

```diff
@@ -185,22 +185,27 @@ def channel_group_search(self, search_term):
         if not isinstance(search_term, str):
             raise TypeError("I can search for str terms only.")
 
-        chg_names = [
-            x.text
-            for x in self._root.findall(".//tdm_channelgroup/name")
-            if x.text is not None
-        ]
+        group_names = {}
+        chg_names = []
+        for x in self._root.findall(".//tdm_channelgroup/name"):
+            if x.text is not None:
+                if x.text not in group_names:
+                    group_names[x.text] = 0
+                else:
+                    group_names[x.text] += 1
+                chg_names.append((f"{x.text}", group_names[x.text]))
```

Comment:
This can just be `x.text`, no?
```suggestion
                chg_names.append((x.text, group_names[x.text]))
```